hadoop-common-project/hadoop-common/src/main/bin/hadoop-functions.sh

```diff
@@ -1569,6 +1569,19 @@ function hadoop_finalize_hadoop_opts
   hadoop_add_param HADOOP_OPTS hadoop.security.logger "-Dhadoop.security.logger=${HADOOP_SECURITY_LOGGER}"
 }
 
+## @description  Finish configuring JPMS that enforced for JDK 17 and higher
+## @description  prior to executing Java
+## @audience     private
+## @stability    evolving
+## @replaceable  yes
+function hadoop_finalize_jpms_opts
+{
+    hadoop_add_param HADOOP_OPTS IgnoreUnrecognizedVMOptions "-XX:+IgnoreUnrecognizedVMOptions"
+    hadoop_add_param HADOOP_OPTS open.sun.security.x509 "--add-opens=java.base/java.util.zip=ALL-UNNAMED"
```

Comment:
Should it be `java.util.zip`?

Reply:
oh, right. it's a copy-paste issue, fixed.